### PR TITLE
[S-02] added devices models with basic endpoints

### PR DESCRIPTION
**Pull request by Riwuko**

### CODE REVIEW

File: devices/api/views.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    permission_classes = [
+        AllowAny,
+    ]
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return self.list_serializer_class
+        return self.serializer_class
+
+    def get_queryset(self):
```
Comment (michalinaasd) [inline devices/api/views.py:34]: Do we need that function? It only call another one.

File: devices/api/views.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        return self.queryset.filter(user=self.request.user)
+
+
+class DeviceViewSet(viewsets.ReadOnlyModelViewSet):
+    queryset = Device.objects.all()
+    serializer_class = DeviceSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(building__user == self.request.user)
```
Comment (michalinaasd) [inline devices/api/views.py:43]: ☝️ same here

File: devices/api/views.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+
+class MeasuringDeviceViewSet(viewsets.ReadOnlyModelViewSet):
+    queryset = MeasuringDevice.objects.all()
+    serializer_class = MeasuringDeviceSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(building__user=self.request.user)
```
Comment (michalinaasd) [inline devices/api/views.py:51]: 👆 same

File: devices/api/views.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+        return Response(response_data, status=status.HTTP_200_OK)
+
+
+class MeasurementViewSet(viewsets.ReadOnlyModelViewSet):
+    queryset = Measurement.objects.all()
+    serializer_class = MeasurementSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(building__user=self.request.user)
```
Comment (michalinaasd) [inline devices/api/views.py:73]: same ☝️

File: devices/models.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
+from django.db import models
+
+from users.models import User
+
+
+class Building(models.Model):
+    name = models.CharField(max_length=100, null=True)
```
Comment (michalinaasd) [inline devices/models.py:7]: Django wtf. Property named null??????????????????????????????

File: devices/models.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    user = models.ForeignKey(
+        User, related_name="user_buildings", null=True, on_delete=models.SET_NULL
+    )
+
+    def __str__(self):
+        return f"Building: {str(self.id)} | name: {self.name}"
+
+
+class Room(models.Model):
+    name = models.CharField(max_length=100, null=False)
```
Comment (michalinaasd) [inline devices/models.py:17]: I think it would be better to store max_length as some Constant structure, because it occures multiple times.

**Review by michalinaasd — CHANGES_REQUESTED** [review]

File: devices/api/views.py (hunk 1 above)
Comment (Riwuko) [reply devices/api/views.py:34]: do u know django? xDD

File: devices/api/views.py (hunk 2 above)
Comment (Riwuko) [reply devices/api/views.py:43]: xDDD

File: devices/models.py (hunk 5 above)
Comment (Riwuko) [reply devices/models.py:7]: it's just field, id is a PK

File: devices/models.py (hunk 6 above)
Comment (Riwuko) [reply devices/models.py:17]: django convention

File: devices/api/views.py (hunk 1 above)
Comment (michalinaasd) [reply devices/api/views.py:34]: nope

**Review by michalinaasd — APPROVED** [review]

**Pull request merged**